README.md

```diff
@@ -1,2 +1,76 @@
-# ts_m2fpga
-M2 Controller FPGA code
+# M2 Cell Controller FPGA Code
+
+## Overview
+
+This project will improve the FPGA code in the [M2 Cell Controller LaVIEW project](https://github.com/lsst-ts/ts_mtm2_cell).
+In that version, the VI named **4-port serial Master Slave.vi**, under `Chassis (cRIO-9038) >> FPGA Target (RIO0, cRIO-9038)` in the Project Explorer, interacts with the hardware inputs and outputs.
+
+## Directory Structure
+
+```bash
+ts_m2fpga
+├── src
+│   ├── m2fpga.lvproj
+│   ├── mainFPGA.vi
+│   ├── moduleRelays.vi
+│   ├── moduleDaq.vi
+│   ├── moduleIlc.vi
+├── doc
+│   ├── versionHistory.md
+│   ├── moduleRelays.md
+│   ├── moduleDaq.md
+│   ├── moduleIlc.md
+├── dependencies
+│   ├── NiFpga_mainFPGA.h
+│   ├── NiFpga_mainFPGA.lvbitx
+│   ├── NiFpga.c
+│   ├── NiFpga.h
+├── README.md
+└── .gitignore
+```
+
+- `src`: in here are all LabVIEW files.
+- `doc`: documentation like `versionHistory.md`.
+- `dependencies`: in here are all files generated by the FPGA Interface C API.
+
+## FPGA Interface C API
+
+The FPGA Interface C API enables C/C++ applications to interact directly with the compiled LabVIEW FPGA VI on CompactRIO devices, generating C functions calls to read from and write to named controls and indicators, perform DMA data transfers, and wait on and acknowledge interrupts.
```

Comment:
on CompactRIO devices --> on real-time controllers

Reply:
Give the full name of "DMA" before the abbreviation.

Reply:
Done